cli/cli_fileshare.go

```diff
@@ -299,6 +328,82 @@ func (c *cmd) FileshareCancel(ctx *cli.Context) error {
 	return nil
 }
 
+// FileshareClear rpc
+func (c *cmd) FileshareClear(ctx *cli.Context) error {
+	if ctx.NArg() < 1 {
+		return formatError(argsCountError(ctx))
+	}
+
+	var resp *pb.Error
+	var err error
+	var until = time.Now()
+
+	args := ctx.Args()
+	if args.Get(0) != "all" {
+		argsJoined := strings.Join(args.Slice(), " ")
+		ts, err := parseTimespan(argsJoined)
+		if err != nil {
+			return formatError(err)
+		}
+		until = until.Add(time.Duration(-1000000000 * ts))
```

Comment:
```suggestion
		until = until.Add(time.Duration(-time.Second * ts))
```

Reply:
I see you are using average values for a month and year which will end up in not correct timestamps. Considering you still need a reference timestamp, which is `var until = time.Now()`, I recommend reworking `parseTimespan`  function so it would return years and months separately. E. g. 
```go
parseTimestamp(input string) (years int, months int, seconds int) { ... }
```
Then you could use a combination of https://pkg.go.dev/time#Time.AddDate and https://pkg.go.dev/time#Time.Add to have correct timestamps